src/main/java/ch/njol/skript/lang/SkriptEvent.java

```diff
@@ -216,6 +216,13 @@ public boolean isEventPrioritySupported() {
 		return true;
 	}
 
+	/**
+	 * Override this method to allow Skript to not force synchronization.
+	 */
+	public boolean allowAsynchronous() {
```

Comment:
I'm a little unsure about the name. What do you think about something like `canExecuteAsynchronously`?

Also, I wonder if it would be a good idea to provide the Event instance here since this method is executed during runtime. It could prove useful for developers.